app/src/main/java/ch/epfl/skysync/models/flight/FinishedFlight.kt

```diff
@@ -17,13 +18,21 @@ data class FinishedFlight(
     override val timeSlot: TimeSlot,
     override val vehicles: List<Vehicle>,
     val color: FlightColor = FlightColor.NO_COLOR,
-    val takeOffTime: LocalTime,
-    val takeOffLocation: Location,
-    val landingTime: LocalTime,
-    val landingLocation: Location,
-    val flightTime: Long // time in milliseconds
+    val takeOffTime: Date,
+    val takeOffLocation: LocationPoint,
+    val landingTime: Date,
+    val landingLocation: LocationPoint,
+    val flightTime: Long, // time in milliseconds
+    val reportId: List<Report> = emptyList()
 ) : Flight {
+
+  private var flightStatus = FlightStatus.MISSING_REPORT
+
   override fun getFlightStatus(): FlightStatus {
-    return FlightStatus.COMPLETED
+    return this.flightStatus
+  }
+
+  fun reportCompleted() {
+    this.flightStatus = FlightStatus.COMPLETED
```

Comment:
make the class immutable to make sure the UI gets noticed of changes.

`getFlightStatus() `does check if 
**number of reports** == **number of team members**

adding a report creates a new FinishedFlight with the reportId added to the list of reports